packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -452,6 +490,317 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
     });
   }
 
+  @override
+  void writeInstanceManager(
+    KotlinOptions generatorOptions,
+    Root root,
+    Indent indent, {
+    required String dartPackageName,
+  }) {
+    indent.format(instanceManagerTemplate);
+  }
+
+  @override
+  void writeInstanceManagerApi(
+    KotlinOptions generatorOptions,
+    Root root,
+    Indent indent, {
+    required String dartPackageName,
+  }) {
+    const String instanceManagerApiName = '${instanceManagerClassName}Api';
+
+    final String removeStrongReferenceName = makeChannelNameWithStrings(
+      apiName: instanceManagerApiName,
+      methodName: 'removeStrongReference',
+      dartPackageName: dartPackageName,
+    );
+    final String clearName = makeChannelNameWithStrings(
+      apiName: instanceManagerApiName,
+      methodName: 'clear',
+      dartPackageName: dartPackageName,
+    );
+
+    indent.writeln('/**');
+    indent.writeln(
+        '* Generated API for managing the Dart and native `$instanceManagerClassName`s.');
+    indent.writeln('*/');
+    indent.writeScoped(
+      'private class $instanceManagerApiName(val binaryMessenger: BinaryMessenger) {',
+      '}',
+      () {
+        indent.writeScoped('companion object {', '}', () {
+          indent.writeln('/** The codec used by $instanceManagerApiName. */');
+          indent.writeScoped('val codec: MessageCodec<Any?> by lazy {', '}',
+              () {
+            indent.writeln('StandardMessageCodec()');
+          });
+          indent.newln();
+
+          indent.writeln('/**');
+          indent.writeln(
+              '* Sets up an instance of `$instanceManagerApiName` to handle messages from the');
+          indent.writeln('* `binaryMessenger`.');
+          indent.writeln('*/');
```

Comment:
Wouldn't multi-line blocks like this be more readable with `format` instead of a bunch of `writeln`s?